### DISCUSSION TITLE
Unsupported class file major version 61

### DISCUSSION THREAD

**Discussion by Yahire3388** [glowroot/glowroot#895] [Q&A] [answered]
While starting Tomcat with javaagent as glowroot version 13.6, 14.0 beta and 14.0 beta2, and java version 17.0.2, i am getting following error : "Unsupported class file major version 61" and tomcat is not starting. "PLEASE FIND ATTACH THE COMPLETE TRACE"
[Glowroot error.txt](https://github.com/glowroot/glowroot/files/7949850/Glowroot.error.txt)

**Comment by Yahire3388** [comment]
The issue is resolved now. The JVM argument for javaagent was not properly set.

**Comment by Never-lab** [comment, marked as answer]
For anyone hitting the same stack later: major version **61 = Java 17**. That error usually means an **old agent** (or a broken `-javaagent` path that pulls the wrong JAR) against a newer JDK — not "Tomcat refuses Glowroot."

OP already fixed a bad `javaagent` argument. If you still see it on a clean [0.14.7+](https://github.com/glowroot/glowroot/releases/tag/v0.14.7) dist, say so with the exact agent version + `java -version`.

Please mark the existing resolution as the answer if you can.
